gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/RedisCache.java

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.node.plugin.cache.redis;
+
+import io.gravitee.node.api.cache.Cache;
+import io.gravitee.node.api.cache.CacheConfiguration;
+import io.gravitee.node.api.cache.CacheException;
+import io.gravitee.node.api.cache.CacheListener;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.core.Single;
+import io.vertx.redis.client.RedisAPI;
+import io.vertx.redis.client.Response;
+import io.vertx.redis.client.ResponseType;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+public class RedisCache<K extends String, V> implements Cache<K, V> {
+
+    public static final String SEPARATOR = ".";
+    private final String name;
+    private final CacheConfiguration cacheConfiguration;
+    private final RedisAPI redisAPI;
+    private final Map<String, CacheListener<K, V>> cacheListeners = new HashMap<>();
+
+    public RedisCache(String name, CacheConfiguration cacheConfiguration, RedisAPI redisAPI) {
```

Comment:
In order to have a better compatibility with generics, maybe consider passing the mapper the constructor instead.